ext/reflection/php_reflection.c

```diff
@@ -2105,7 +2127,108 @@ ZEND_METHOD(ReflectionFunctionAbstract, getNumberOfRequiredParameters)
 }
 /* }}} */
 
-/* {{{ Returns an array of parameter objects for this function */
+/* {{{ Returns whether a parameter exists or not */
+ZEND_METHOD(ReflectionFunctionAbstract, hasParameter)
+{
+    reflection_object *intern;
+    zend_function *fptr;
+    zend_string *arg_name = NULL;
+    zend_long position;
+    uint32_t num_args;
+
+    ZEND_PARSE_PARAMETERS_START(1, 1)
+        Z_PARAM_STR_OR_LONG(arg_name, position)
+    ZEND_PARSE_PARAMETERS_END();
+
+    GET_REFLECTION_OBJECT_PTR(fptr);
+
+    num_args = fptr->common.num_args;
+
+    if (fptr->common.fn_flags & ZEND_ACC_VARIADIC) {
+        num_args++;
+    }
+
+    if (!num_args) {
+        RETURN_FALSE;
+    }
+
+    if (arg_name != NULL) {
+        if (get_parameter_position(fptr, arg_name, num_args) > -1) {
+            RETURN_TRUE;
+        }
+
+        RETURN_FALSE;
+    } else {
+        if (position < 0) {
+            zend_argument_value_error(1, "must be greater than or equal to 0");
+            RETURN_THROWS();
+        }
+
+        RETURN_BOOL(position < num_args);
+    }
+}
+/* }}} */
+
+/* {{{ Returns a parameter specified by its name */
+ZEND_METHOD(ReflectionFunctionAbstract, getParameter)
+{
+    reflection_object *intern;
+    zend_function *fptr;
+    zend_string *arg_name = NULL;
+    zend_long position;
+    struct _zend_arg_info *arg_info;
+    uint32_t num_args;
+    zval reflection;
+
+    ZEND_PARSE_PARAMETERS_START(1, 1)
+        Z_PARAM_STR_OR_LONG(arg_name, position)
+    ZEND_PARSE_PARAMETERS_END();
+
+    GET_REFLECTION_OBJECT_PTR(fptr);
+
+    num_args = fptr->common.num_args;
+    arg_info = fptr->common.arg_info;
+
+    if (fptr->common.fn_flags & ZEND_ACC_VARIADIC) {
+        num_args++;
+    }
+
+    if (!num_args) {
+        RETURN_THROWS();
+    }
+
+    if (arg_name != NULL) {
+        position = get_parameter_position(fptr, arg_name, num_args);
+
+        if (position == -1) {
+            _DO_THROW("The parameter specified by its name could not be found");
+            RETURN_THROWS();
+        }
+    } else {
```

Comment:
Ditto check for empty string?

Reply:
> Ditto check for empty string?

Is this just as simple as `arg_name != ''` or is there a specific way I need to handle it @Girgias ?

Reply:
Check that the string lengths is not 0 :)

``ZSTR_LEN(arg_nam) != 0``